src/backend/distributed/master/worker_node_manager.c

```diff
@@ -332,6 +332,12 @@ ActivePrimaryNodeList(void)
 	HTAB *workerNodeHash = GetWorkerNodeHash();
 	HASH_SEQ_STATUS status;
 
+	if (ReadFromSecondaries == READ_FROM_SECONDARIES_ALWAYS)
+	{
+		ereport(ERROR, (errmsg("writing to nodes is not currently allowed"),
+						errdetail("citus.read_from_secondaries is set to 'always'")));
```

Comment:
I'm starting to think you're right but there's one weird behavior: Say that, if you're on a primary coordinator you can continue to do all the things you could previously do. This means once you flip `read_from_secondaries` all your reads go to different nodes than your writes.